modules.d/00systemd/module-setup.sh

```diff
@@ -1,10 +1,14 @@
 #!/bin/bash
 
+getSystemdVersion() {
+    $systemdutildir/systemd --version | { read a b a; echo $b; }
+}
+
 # called by dracut
 check() {
     [[ $mount_needs ]] && return 1
     if require_binaries $systemdutildir/systemd; then
-        SYSTEMD_VERSION=$($systemdutildir/systemd --version | { read a b a; echo $b; })
+        SYSTEMD_VERSION=$(getSystemdVersion)
         # Check if the systemd version is a valid number
         if ! [[ $SYSTEMD_VERSION =~ ^[0-9]+$ ]]; then
```

Comment:
Just a one nitpick - could you move this check into the `getSystemdVersion` function as well, so it's checked in every call?